gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DbDataStore.js

```diff
@@ -0,0 +1,599 @@
+/*!
+GPII CouchDB OAuth 2 Data Store
+
+Copyright 2016 OCAD university
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = fluid || require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+require("./DbDataStoreUtils.js");
+
+fluid.defaults("gpii.oauth2.dbDataSource", {
+    gradeNames: ["kettle.dataSource.URL", "kettle.dataSource.CouchDB"],
+    baseUrl: null,   // Supplied by integrators
+    termMap: {
+        dbName: null    // Supplied by integrators
+    },
+    notFoundIsEmpty: true,
+    rules: {
+        writePayload: {
+            "": ""
+        },
+        readPayload: {
+            "": ""
+        }
+    },
+    // Add baseUrl value into url template since baseUrl (an example baseUrl value: http://localhost:5984/)
+    // should not be passed into the "termMap" option because those termMap values will be encoded to
+    // compose the url.
+    url: {
```

Comment:
Rewrite this using the standard URL interpolation scheme for dataSources and the "noencode:" directive